TypeCobol.Test/Utils/Comparator.cs

```diff
@@ -49,11 +49,26 @@ public void Init(string[] extensions = null, bool autoRemarks = false, bool Antl
         }
 
 		public void Parse() {
-			try { Compiler.CompileOnce(); }
+#if EUROINFO_RULES
+			try
+            {
+                //Check for the presence of a CpyCopies.lst file.
+                FileInfo fi = new FileInfo(this.Compiler.CobolFile.FullPath);
+                FileInfo copyFileMap = new FileInfo(Path.Combine(fi.DirectoryName, fi.Directory.Name + "CpyCopies.lst"));
+                if (copyFileMap.Exists) { 
+                    Compiler.CompilerOptions.CpyCopyNamesMap = new TypeCobol.Compiler.Preprocessor.CopyNameMapFile(copyFileMap.FullName);
```

Comment:
Initialization of `Compiler.CompilerOptions.CpyCopyNamesMap` could be moved into `Init` method just above.
And you don't need to set `Compiler.CompilerOptions.CpyCopyNamesMap = null;` as it's not a Singleton anymore.
So the method `Parse` can remains unchanged.

Reply:
No because this is test and dirctory dependent. and it applies at compile time.